File: 5.Caesar/caesar.py
#!/usr/bin/env python3

from sys import argv
from caesar_class import Caesar


def main():

    if not argv_is_valid(argv):
        exit("Usage: python caesar.py key")

    key = argv[1]
    caesar = Caesar(int(key))
    plain_text = input("plaintext: ")
    print("ciphertext: ", end="")

    for char in plain_text:
        print(caesar.shift_by_key(char), end="")
    print()


def get_ascii_value(self, character):
    """Returns the ASCII value of the character provided."""

    return ord(character)


def argv_is_valid(argv_list):
    """Returns whether or not the commandline arguments provided are valid."""

    try:
        correct_length = len(argv_list) == 2
        second_arg_is_numeric = argv_list[1].isnumeric()
        return correct_length and second_arg_is_numeric
    except IndexError:
        return False


if __name__ == "__main__":
    main()

File: 2.C++_solutions/5.Caesar/Caesar.cpp
#include <cctype>
#include <iostream>
#include <string>

using namespace std;

bool onlyDigits(int argc, char *argv[]);

int main(int argc, char *argv[])
{
    if (!(onlyDigits(argc, argv)))
    {
        cout << "Usage: ./caesar key";
        return 1;
    }
    
    return 0;

}

// Returns whether the command-line arguments provided
// are valid.
bool onlyDigits(int argc, char *argv[])
{
    if (!(argc == 2))
        return false;
    
    // Check that the key supplied is numeric, by checking every character of the key.
    string key = argv[1];
    string::size_type length = key.length();

    for (int i = 0; i < static_cast<int>(length); i++)
        if (!isdigit(key[i]))
            return false;
    
    return true;
}